File: csp/cspsim/windows/MainMenu.cpp
// Combat Simulator Project
// Copyright (C) 2002 The Combat Simulator Project
// http://csp.sourceforge.net
//
// This program is free software; you can redistribute it and/or
// modify it under the terms of the GNU General Public License
// as published by the Free Software Foundation; either version 2
// of the License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program; if not, write to the Free Software
// Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.


/**
 * @file MainMenu.cpp
 *
 **/

#include <csp/csplib/data/DataManager.h>
#include <csp/cspsim/CSPSim.h>
#include <csp/cspsim/DynamicObject.h>
#include <csp/cspsim/battlefield/LocalBattleField.h>
#include <csp/cspsim/wf/Button.h>
#include <csp/cspsim/wf/Serialization.h>
#include <csp/cspsim/wf/WindowManager.h>
#include <csp/cspsim/windows/MainMenu.h>
#include <csp/cspsim/windows/Options.h>

CSP_NAMESPACE

namespace windows {

MainMenu::MainMenu() {
	Ref<wf::Serialization> serializer = new wf::Serialization(getUIPath());
	serializer->load(this, getUITheme(), "main_menu.xml");

	Ref<wf::Button> instantActionButton = getById<wf::Button>("instantAction");
	if(instantActionButton.valid())
		instantActionButton->addButtonClickedHandler(sigc::mem_fun(*this, &MainMenu::instantAction_Click));

	Ref<wf::Button> optionsButton = getById<wf::Button>("options");
	if(optionsButton.valid())
		optionsButton->addButtonClickedHandler(sigc::mem_fun(*this, &MainMenu::options_Click));

	Ref<wf::Button> quitButton = getById<wf::Button>("quit");
	if(quitButton.valid())
		quitButton->addButtonClickedHandler(sigc::mem_fun(*this, &MainMenu::quit_Click));	
}

void createVehicleHelper(CSPSim *self, const char *path, Vector3 position,
                          Vector3 velocity, Vector3 attitude) {
	Ref<DynamicObject> obj = self->getDataManager().getObject(path);
	if (!obj) {
		return;
	}
	
	obj->setGlobalPosition(position);
	obj->setVelocity(velocity);
	csp::Quat q_attitude;
	attitude *= 3.1416 / 180.0;
	q_attitude.makeRotate(attitude.x(), attitude.y(), -attitude.z());
	obj->setAttitude(q_attitude);
	self->getBattlefield()->__test__addLocalHumanUnit(obj);
	if (!self->getActiveObject()) self->setActiveObject(obj);
}


void MainMenu::instantAction_Click() {
	CSPSim::theSim->displayLogoScreen();
	CSPSim::theSim->loadSimulation();

	createVehicleHelper(CSPSim::theSim, "sim:vehicles.aircraft.f16dj", Vector3(-29495, -10530, 91.1), Vector3(0, 0, 0), Vector3(0.0, 0.0, 180.0));
	createVehicleHelper(CSPSim::theSim, "sim:vehicles.aircraft.f16dj", Vector3(-29510, -10530, 91.1), Vector3(0, 0, 0), Vector3(0.0, 0.0, 180.0));
}

void MainMenu::quit_Click() {
	CSPSim::theSim->quit();
}

void MainMenu::options_Click() {
	wf::WindowManager* manager = getWindowManager();
	if(manager != NULL) {
		if(!manager->windowIsOpen<Options>()) {
			Ref<Window> options = new Options();
			manager->show(options.get());
			options->centerWindow();
		}
	}
}

MainMenu::~MainMenu() {
}

} // namespace windows

CSP_NAMESPACE_END

File: csp/data/ui/scripts/windows/mainmenu.py
#!/usr/bin/python

# Combat Simulator Project
# Copyright (C) 2002-2005 The Combat Simulator Project
# http://csp.sourceforge.net
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.

"""
Combat Simulator Project : Main menu script
"""

import csp.csplib
from csp.data.ui.scripts.gamescreenmanager import GameScreenManager
from csp.data.ui.scripts.utils import SlotManager
from csp.data.ui.scripts.windows.options import Options
from csp.data.ui.scripts.windows.tutorials import Tutorials

class MainMenu(csp.cspsim.Window, SlotManager):
    def __init__(self, cspsim):
        csp.cspsim.Window.__init__(self)
        SlotManager.__init__(self)

        self.cspsim = cspsim
        
        serializer = csp.cspsim.Serialization()
        serializer.load(self, 'main_menu.xml')
        
        instantActionButton = self.getById('instantAction')
        if instantActionButton != None:
            self.connectToClickSignal(instantActionButton, self.instantAction_Click)
        
        optionsButton = self.getById('options')
        if optionsButton != None:
            self.connectToClickSignal(optionsButton, self.options_Click)
            
        tutorialsButton = self.getById('tutorials')
        if tutorialsButton != None:
            self.connectToClickSignal(tutorialsButton, self.tutorials_Click)
        
        quitButton = self.getById('quit')
        if quitButton != None:
            self.connectToClickSignal(quitButton, self.quit_Click)

    def instantAction_Click(self):
        self.getWindowManager().closeAll()
        self.cspsim.displayLogoScreen()
        self.cspsim.loadSimulation()
        
        vec = csp.csplib.Vector3
        m2k = "sim:vehicles.aircraft.m2k"
        f16dj = "sim:vehicles.aircraft.f16dj"
        self.cspsim.createVehicle(f16dj, vec(-29495, -10530, 91.3), vec(0, 0, 0), vec(0.0, 0.0, 180.0), True) 
        self.cspsim.createVehicle(f16dj, vec(-29510, -10530, 91.3), vec(0, 0, 0), vec(0.0, 0.0, 180.0), False) 
        
        gameScreenManager = GameScreenManager(self.cspsim)
        
    def tutorials_Click(self):
        tutorials = Tutorials(self.cspsim)
        tutorials.displayMission('index.xml')
        self.getWindowManager().show(tutorials)
                        
    def options_Click(self):
        options = Options(self.cspsim)
        self.getWindowManager().show(options)
        
    def quit_Click(self):
        self.cspsim.quit()
        
